_docs/development/xampp_php_versionen.md

```diff
@@ -0,0 +1,630 @@
+---
+title: XAMPP mit mehreren PHP-Versionen (Windows)
+authors: [aeberhard]
+prio:
+---
+
+[ letztes update: 2022-12-25 ]
+
+# XAMPP mit mehreren PHP-Versionen (Windows)
+
+Unter Windows ist m. E. [**XAMPP**](https://www.apachefriends.org/) (Apache + MariaDB + PHP + Perl) die einfachste Basis für die REDAXO-Entwicklung. Ich selbst (@aeberhard) verwende XAMPP schon seit sehr vielen Jahren.
+
+Hier eine Anleitung um **XAMPP** mit mehreren PHP-Versionen und SSL-Verschlüsselung einzurichten!
+
+> **Hinweis:** Bei Installation in einem anderen Ordner als `c:\work` müssen die Pfade entsprechend angepasst werden.
+
+- [XAMPP installieren](#xamppinstall)
+- [XAMPP Setup](#xamppsetup)
+- [Anpassung der php.ini](#xamppphpini)
+- [SSL Zertifikat einrichten](#xamppssl)
+- [Zusätzliche PHP-Versionen installieren](#xamppphpversions)
+- [Apache konfigurieren](#xamppapache)
+- [REDAXO installieren](#xamppredaxo)
+- [Sonstiges / Hinweise](#xampphints)
+
+---
+
+<a name="xamppinstall"></a>
+
+## XAMPP installieren
+
+**Download** unter folgender URL: [https://sourceforge.net/projects/xampp/files/XAMPP%20Windows/](https://sourceforge.net/projects/xampp/files/XAMPP%20Windows/)
+
+Hier die gewünschte PHP-Version auswählen und eine `portable` ZIP-Version verwenden.
+
+> **Hinweis:** Beispielhaft wird hier die letzte PHP 7.4.33 Version verwendet und im Ordner `c:\work\xampp` installiert. Diese PHP-Version ist dann auch die Standard-Version des lokalen XAMPP.
+
+**Download-Link:** [https://sourceforge.net/projects/xampp/files/XAMPP%20Windows/7.4.33/](https://sourceforge.net/projects/xampp/files/XAMPP%20Windows/7.4.33/)
+
+Screenshot:
+
+![XAMPP Download](https://raw.githubusercontent.com/FriendsOfREDAXO/tricks/master/screenshots/xampp/XAMPP-Download.png "XAMPP Download")
+
+> **Hinweis:** Download Visual C++ Redistributable für Visual Studio 2015 falls noch nicht auf dem System vorhanden [https://www.microsoft.com/de-de/download/details.aspx?id=48145](https://www.microsoft.com/de-de/download/details.aspx?id=48145)
+
+### Installation
+
+1. XAMPP-Archiv in den gewünschten Ordner `c:\work` herunterladen
+2. XAMPP-Archiv entpacken
+3. In den entpackten Ordner `C:\Work\xampp-portable-windows-x64-7.4.33-0-VC15` wechseln
+4. Das Verzeichnis `xampp` **ausschneiden** und unter `c:\work` wieder **einfügen**
+5. Ordner `C:\Work\xampp-portable-windows-x64-7.4.33-0-VC15` kann jetzt gelöscht werden
```

Comment:
Was ist denn da gängige Empfehlung? Ich habe den bei mir das in `C:\xampp\` liegen, aber vom Hörensagen und Halbwissen könnte es sein, dass der Benutzer-AppData oder Dokumenten-Ordner besser wäre. Ein eigenes Verzeichnis auf C:\ sollte man ja wenn möglich vermeiden.

Reply:
Ich habe ein Verzeichnis `C:\Daten` und da dann alles mögliche drin.
z.B. ein Unterverzeichnis `Software` in dem alle portablen Programme und auch mehrere xampp's drinliegen.
Evtl. ein prominenterer Hinweis dass das `work` nur ein Beispiel ist und der eigene Pfad verwendet werden kann!?

Reply:
Der Vorteil von `c:\work\xampp` ist dass das Verzeichnis auch auf einen USB-Stick kopiert werden kann.
Sieht halt besser aus als dann ein Verzeicnis `\Users\userid\AppData\xampp` auf dem Stick zu haben.

Reply:
Ich habe den Kram auch auf d:\xampp

Reply:
Ich habe nochmal nachgelesen und dann ist es mir auch wieder eingefallen. Wenn's nicht im Benutzer-Ordner installiert wird, sondern so wie von xampp vorgegeben oder in deinem Beispiel, dann stürzt xampp beim schreiben der Konfigurationsdatei config.ini ab.

Das ist ein Problem mit fehlenden Schreibrechten, die man nur hat, weil xampp bis heute nichts von der in Windows Vista eingeführten Benutzerkonto-Steuerung weiß.

Im letzten Kommentar hier wird's ganz gut erklärt. https://www.drwindows.de/xf/threads/xampp-control-panel-schreibrechte-einr%C3%A4umen.136650/

Ich führe deshalb xampp als Admin aus, um das zu umgehen, finde das aber auch nicht optimal.

Aber das nur nebenbei. Der Standard-Pfad ist C:\xampp also würde ich den auch hier nehmen.

Reply:
Ich werde das jetzt mal anpassen auf D:\xampp mit einem Hinweis nicht in C:\xampp zu installieren :-)